src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray`1.cs

```diff
@@ -726,9 +730,13 @@ public ImmutableArray<T> Remove(T item, IEqualityComparer<T> equalityComparer)
         {
             var self = this;
             self.ThrowNullRefIfNotInitialized();
+            
+            if (self.IsEmpty)
+                return self;
+            
```

Comment:
Is it common to Remove from an empty list?  Upfront checks like this that optimize for the empty case are great if we expect it to be empty, but if it's more likely for it not to be empty, then this actualy penalizes the common case.